data/transactions/logic/assembler.go

```diff
@@ -1399,25 +1399,26 @@ func typecheck(expected, got StackType) bool {
 	return expected == got
 }
 
-var spaces = [256]uint8{'\t': 1, ' ': 1}
+// semi-colon is quite space-like, so include it
+var spaces = [256]bool{'\t': true, ' ': true, ';': true}
```

Comment:
I like this change, but I will say it does mean that we lose the ability to make some weird macros like a macro beginning with a semicolon as we will now separate the semicolon and macro into two distinct tokens. I think that's fine but just something to keep track of

Reply:
Worth keeping in mind, but I can't think of an macro that would want to terminate the last "line" from the beginning.

Reply:
Oh, but it might not be as bad as you're thinking. Although ";" is it's own token, since "# directives" will get the whole rest of the line, I think "#define x y; z" will still cause x to become "x ; z"